web3pi_proxy/core/rpc/node/endpoint_pool/pool_manager.py

```diff
@@ -92,14 +90,114 @@ def check_connections(self, pools: List[EndpointConnectionPool]):
                     daemon=True,
                 ).start()
 
-            if self.__is_outdated(pool):
-                self.__logger.warning(f"Endpoint `{pool.endpoint.name}`: falling behind")
-
             self.__logger.debug(f"Endpoint `{pool.endpoint.name}`: okay.")
 
 
+class HttpResponseParserListenerSyncController:
+    body: bytes = b''
+    block_number: int = 0
+    block_timestamp: int = 0
+    __logger = get_logger("SyncController")
+
+    def __init__(self, __pool_name: str):
+        super().__init__()
+        self.__pool_name = __pool_name
+
+    def on_body(self, body: bytes):
+        self.body = self.body + body
+
+    def on_message_complete(self):
+        try:
+            block_data = json.loads(self.body)
+            result = block_data.get('result')
+            if not result:
+                error = block_data.get('error')
+                if error:
+                    self.__logger.error(f"{self.__pool_name}: Sync test failed: {error}")
+                else:
+                    self.__logger.error(f"{self.__pool_name}: Sync test failed: reason unknown")
+                return
+            self.block_number = int(result['number'], 16)
+            self.block_timestamp = int(result['timestamp'], 16)
+        except Exception as error:
+            self.__logger.error("%s: %s", error.__class__, error)
+            self.__logger.error(f"{self.__pool_name}: Failed to failed to parse sync test response")
+
+
+# TODO: Tune parameters
+class SyncController:
+    __SUSPENSION_TIMEOUT_SECONDS = 60
+    __MAX_DELAY_SECONDS = 60
+
+    __logger = get_logger("SyncController")
+
+    def __test_pool(self, pool: EndpointConnectionPool) -> bool | None:
+        endpoint_connection_handler: EndpointConnectionHandler | None = None
+        try:
+            endpoint_connection_handler = pool.get(out_of_sync=True)
+            req = RPCRequest()
+            req.headers = b'User-Agent: curl/8.0.1\r\nAccept: */*\r\nContent-Type: application/json\r\nContent-Length: 82\r\n'
+            req.content = b'{"method":"eth_getBlockByNumber","params":["latest",false],"id":1,"jsonrpc":"2.0"}'
```

Comment:
```suggestion
            req = RPCRequest(
                headers = b'User-Agent: curl/8.0.1\r\nAccept: */*\r\nContent-Type: application/json\r\nContent-Length: 82\r\n'
                content = b'{"method":"eth_getBlockByNumber","params":["latest",false],"id":1,"jsonrpc":"2.0"}'
            )
```

but even better, I'd love some helper method, e.g. `RPCRequest.new()` that would take data as a dictionary and would construct the resultant `RPCRequest` object with properly initialized `headers` and `content` without having to manually submit bytestrings :/